### PR TITLE
[Feature][connector common] Add Hadoop2 and Hadoop3 shade jar

### PR DESCRIPTION
**Pull request by EricJoy2048**
It will conflict when we added Hadoop dependency in new connectors. So I add Hadoop2 shade jar and Hadoop3 shade jar.

### CODE REVIEW

**Comment by Hisoka-X** [comment]
There have a problem, can't build hadoop 2.x pass on jdk 11. Because hadoop 2.x use jdk tool.jar was removed in JDK 9. Check this: https://stackoverflow.com/questions/53707666/how-to-get-tools-jar-for-openjdk-11-on-windows. Any suggestion? @ruanwenjun @CalvinKirs

**Comment by EricJoy2048** [comment]
> There have a problem, can't build hadoop 2.x pass on jdk 11. Because hadoop 2.x use jdk tool.jar was removed in JDK 9. Check this: https://stackoverflow.com/questions/53707666/how-to-get-tools-jar-for-openjdk-11-on-windows. Any suggestion? @ruanwenjun @CalvinKirs

Can this problem be solved through profile?
maven profile can activation by jdk version. 
For example:

```
<profiles>  
       <profile>  
              <id>onlyhadoop3</id>  
              <jdk>[1.9,1.12)</jdk>  
       </profile>  
<profiles>  
```

**Comment by EricJoy2048** [comment]
maven have some parameter can custom build behavior :
`-am --also-make` At the same time, build the dependent modules of the listed modules;
`-amd -also-make-dependents` At the same time, build modules that depend on the listed modules;
`-pl --projects <arg>` Build the specified modules, which are separated by commas;
`-rf -resume-from <arg>` Restore the reactor from the designated module.

For jdk version > 1.9 we can make all connectors in the project only depend on` seatunnel-hadoop3-shade` with `profile`, and the dependency on `seatunnel-hadoop2-shade` will become 0. Then, we can skip the build of `seatunnel-hadoop2-shade` by using the `mvn XXX -am` command.

**Comment by ruanwenjun** [comment]
It seems we can only use profile to deal with different JDK, although this will make thing conplex.

BTW, the module name `seatunnel-connectors-common` is confusion, I don't think we need to add hadoop dependency in the common mudule.

File: seatunnel-connectors/seatunnel-connectors-seatunnel/seatunnel-connectors-common/seatunnel-hadoop2-shade/pom.xml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                    <execution>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>shade</goal>
+                        </goals>
+                        <configuration>
+                            <relocations>
+                                <relocation>
+                                    <pattern>org.apache.hadoop</pattern>
+                                    <shadedPattern>seatunnel.org.apache.hadoop</shadedPattern>
```
Comment (CalvinKirs) [inline seatunnel-connectors/seatunnel-connectors-seatunnel/seatunnel-connectors-common/seatunnel-hadoop2-shade/pom.xml]: it's better to exclude some  `META-INF` files.

**Comment by CalvinKirs** [comment]
BTW, you should test in your local env, I think some problems don't resolve.

**Comment by Hisoka-X** [comment]
Another problem: when build on jdk11 and remove hadoop2. The build will pass, but license check will failed.Because our license have hadoop2, but build doesn't have it.

**Comment by Hisoka-X** [comment]
In my view, the better way is remove build on jdk11

**Comment by EricJoy2048** [comment]
> It seems we can only use profile to deal with different JDK, although this will make thing conplex.
> 
> BTW, the module name `seatunnel-connectors-common` is confusion, I don't think we need to add hadoop dependency in the common mudule.

OK, what name do you think is more appropriate?

**Comment by Hisoka-X** [comment]
> > It seems we can only use profile to deal with different JDK, although this will make thing conplex.
> > BTW, the module name `seatunnel-connectors-common` is confusion, I don't think we need to add hadoop dependency in the common mudule.
> 
> OK, what name do you think is more appropriate?

I had change it to `seatunnel-connector-hadoop-shade`, is it ok?

**Comment by EricJoy2048** [comment]
> > > It seems we can only use profile to deal with different JDK, although this will make thing conplex.
> > > BTW, the module name `seatunnel-connectors-common` is confusion, I don't think we need to add hadoop dependency in the common mudule.
> > 
> > 
> > OK, what name do you think is more appropriate?
> 
> I had change it to `seatunnel-connector-hadoop-shade`, is it ok?

I am afraid we need more shade jar in the feature. like hadoop2-shade, hadoop3-shade, hive1.2-shade, hive2-shade, hive3-shade ... xxx-shade . So I think we should put this shade modules into a same parent module.

**Comment by Hisoka-X** [comment]
> > > > It seems we can only use profile to deal with different JDK, although this will make thing conplex.
> > > > BTW, the module name `seatunnel-connectors-common` is confusion, I don't think we need to add hadoop dependency in the common mudule.
> > > 
> > > 
> > > OK, what name do you think is more appropriate?
> > 
> > 
> > I had change it to `seatunnel-connector-hadoop-shade`, is it ok?
> 
> I am afraid we need more shade jar in the feature. like hadoop2-shade, hadoop3-shade, hive1.2-shade, hive2-shade, hive3-shade ... xxx-shade . So I think we should put this shade modules into a same parent module.

Or call `seatunnel-connector-shade` ？

**Comment by Hisoka-X** [comment]
The build pass, If don't have other problem, I will merge this first.

**Pull request merged**